spa/src/components/portal/ContributionsList/ContributionDetail/BillingDetails/BillingDetails.tsx

```
@@ -27,18 +36,99 @@ export const INTERVAL_NAMES: Record<ContributionInterval, string> = {
 
 // This will eventually have an editable state.
 
-export function BillingDetails({ contribution, disabled }: BillingDetailsProps) {
+export function BillingDetails({
+  contribution,
+  disabled,
+  editable,
+  onEdit,
+  onEditComplete,
+  onUpdateBillingDetails
+}: BillingDetailsProps) {
+  const amountInDollars = useMemo(() => contribution.amount / 100, [contribution.amount]);
+  const [amount, setAmount] = useState(amountInDollars.toString());
+
+  const showEditControls = useMemo(
+    () => contribution.is_modifiable && contribution.revenue_program.organization.plan.name === PLAN_NAMES.PLUS,
+    [contribution.is_modifiable, contribution.revenue_program.organization.plan.name]
+  );
+
   const formattedDate = Intl.DateTimeFormat(undefined, { day: 'numeric', month: 'long', year: 'numeric' }).format(
     // TODO in DEV-5138: use only first_payment_date
     new Date(contribution.first_payment_date ?? contribution.created)
   );
 
+  const disableSave = useMemo(() => {
+    const onlyZeroOrDotRegex = /^[0.]+$/;
+
+    const isSame = amount === amountInDollars?.toString();
+    const isEmpty = !amount;
+    const isOnlyZero = !!onlyZeroOrDotRegex.exec(amount);
+
+    const parsedValue = parseFloatStrictly(amount);
+    const isValid = !isNaN(parsedValue) && parsedValue > 0;
+
+    return isSame || isEmpty || isOnlyZero || !isValid;
+  }, [amount, amountInDollars]);
+
+  const handleSave = () => {
+    // Amount will be a valid number at this point and we need to convert it to cents
```

Let's throw an error if this isn't true, just in case. I think throwing is appropriate here because we don't want to put bad data into the backend. Doesn't need test coverage since this should be an impossible condition.

---

[7651576](https://github.com/newsrevenuehub/rev-engine/pull/1625/commits/76515762ab213924e73bdbc66a8be458453a81fb)